Kernel/FileSystem/Ext2Base.h

```diff
@@ -0,0 +1,312 @@
+// Group descriptor sizes
+constexpr size_t min_group_descriptor_size       = 32,
+                 min_group_descriptor_size_64bit = 64,
+                 max_group_descriptor_size       = min_group_descriptor_size;
+
+// Block sizes
+constexpr size_t min_block_log_size = 10,
+                 max_block_log_size = 16,
+                 min_block_size = (1 << min_block_log_size),
+                 max_block_size = (1 << max_block_log_size);
+
+// Fragment sizes
+constexpr size_t min_fragment_size = min_block_size,
+                 max_fragment_size = max_block_size,
+                 min_fragment_log_size = min_block_log_size;
+
+// The size of an inode
+constexpr size_t good_old_inode_size = 128;
+
+// This is how many hard links are allowed at maximum.
+constexpr u32 link_max = 65000;
+
+enum class FeatureFullCompat : u32 {
+    DirPrealloc   = (1 << 0),
+    ImagicInodes  = (1 << 1),
+    HasJournal    = (1 << 2),
+    ExtendedAttrs = (1 << 3),
+    ResizeInode   = (1 << 4),
+    DirIndex      = (1 << 5),
+};
+
+enum class FeatureReadOnlyCompat {
+    SparseSuper = (1 << 0),
+    LargeFile   = (1 << 1),
+};
+
+AK_ENUM_BITWISE_OPERATORS (FeatureFullCompat);
+AK_ENUM_BITWISE_OPERATORS (FeatureReadOnlyCompat);
+
+// Special inode numbers
+enum class InodeNums : u64 {
+    Bad = 1,         // Bad blocks inode
+    Root,            // Root inode
+    ACLIndex,        // ACL inode
+    ACLData,         // ACL inode
+    BootLoader,      // Boot loader inode
+    UndeleteDir,     // Undelete directory inode
+    Resize,          // Reserved group descriptors inode
+    Journal,         // Journal inode
+    
+    FirstInode = 11, // First usable inode
+};
+
+// File system states
+enum class FileSystemState : u32 {
+    Valid,  // Cleanly unmounted.
+    Error,  // Errors detected
+    Orphan, // EXT3 Orphans being recovered
+};
+
+// Structure of a block group descriptor
+struct GroupDescriptor {
+    u32 bg_block_bitmap;      // Blocks bitmap block
+    u32 bg_inode_bitmap;      // Inodes bitmap block
+    u32 bg_inode_table;       // Inodes table block
+    u16 bg_free_blocks_count; // Free blocks count
+    u16 bg_free_inodes_count; // Free inodes count
+    u16 bg_used_dirs_count;   // Directories count
+    u16 bg_flags;
+    u32 bg_reserved[2];
+    u16 bg_itable_unused; // Unused inodes count
+    u16 bg_checksum;      // crc16(volume_uuid+group_num+group_desc)
+};
+
+// Structure of an inode on the disk
+struct Inode {
+    u16 mode;        // File mode
+    u16 owner_uid;   // Low 16 bits of Owner Uid
+    u32 file_size;   // Size in bytes
+    u32 access_time; // Access time
+    u32 create_time; // Inode change time
+    u32 modify_time; // Modification time
+    u32 delete_time; // Deletion Time
+    u16 group_id;    // Low 16 bits of Group Id
+    u16 num_links;   // Links count
+    u32 num_blocks;  // Blocks count
+    u32 file_flags;  // File flags
+    
+    // Linux-dependent on
+    u32 linux_version;
+    // Linux-dependent off
```

Comment:
I think that's overkill; just a leading comment "this is linux dependent" should be enough. Below as well.